tests/CMakeLists.txt

```diff
@@ -32,42 +32,56 @@ add_compile_definitions(BUILD_TESTING_EVSE_SECURITY)
 
 add_test(${TEST_TARGET_NAME} ${TEST_TARGET_NAME})
 
-install(
-    PROGRAMS "${CMAKE_CURRENT_SOURCE_DIR}/generate_test_certs.sh"
-    PROGRAMS "${CMAKE_CURRENT_SOURCE_DIR}/generate_test_certs_multi.sh"
+setup_target_for_coverage_gcovr_html(
+    NAME ${PROJECT_NAME}_gcovr_coverage
+    EXECUTABLE ctest
+    DEPENDENCIES ${TEST_TARGET_NAME}
+    EXCLUDE "tests/*"
+)
+
+setup_target_for_coverage_gcovr_xml(
+    NAME ${PROJECT_NAME}_gcovr_coverage_xml
+    EXECUTABLE ctest
+    DEPENDENCIES ${TEST_TARGET_NAME}
+    EXCLUDE "tests/*"
+)
+
+file(COPY
+    "${CMAKE_CURRENT_SOURCE_DIR}/generate_test_certs.sh"
+    "${CMAKE_CURRENT_SOURCE_DIR}/generate_test_certs_multi.sh"
     DESTINATION "${CMAKE_BINARY_DIR}/tests"
 )
 
-install(
-    DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/configs"
+file(COPY
```

Comment:
Installing these files during configure makes the CI reusable workflow happy. Was there a reason why they were installed before? @james-ctc

Reply:
install was being used in other places so it was done to follow the same pattern.

Not a problem here, but in some tests there were issues that the file(COPY ...) doesn't appear to be part of the dependencies so when there is an update ninja/make doesn't copy the updates. It is possible that was the original reason for using install.

Elsewhere I have added specific targets where checking for updates is important.

moving away from using install is a very good improvement.